### ISSUE TITLE
Directory.EnumerateFiles fails to trim space on directory name, returns invalid file paths

### ISSUE THREAD

**Issue by IanMercer** [dotnet/runtime#37212]
`Directory.Enumerate` files concatenates the directory name and the found file name but does so without first trimming whitespace off the directory name. So instead of failing with an error or finding nothing it returns invalid file name strings like "C:\ \pagefile.sys" with a back slash, a space and another backslash.

I think the issue is [here](https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.cs#L44) where it uses the original, untrimmed directory name. 

        [Test]
        public void CanFindFilesWithSpaceOnEndOfDirectory()
        {
            string path = "C:\\ ";    // note: this has a space on the end of the path
            var pagefile = System.IO.Directory.EnumerateFiles(path, "pagefile.sys").First();
            pagefile.Should().Be("C:\\pagefile.sys");
        }

Results:

       Expected pagefile to be 
         "C:\pagefile.sys" with a length of 15, but 
         "C:\ \pagefile.sys" has a length of 17, differs near " \p" (index 3).

This is a .NETStandard 2.0 project building on VS 16.6.0 compiling to x64 but I don't think it's specific to that environment.

**Comment by Clockwork-Muse** [comment]
It's possible to create directories with trailing whitespace (say, using WSL):

![image](https://user-images.githubusercontent.com/5569508/83358566-467ff900-a329-11ea-9918-a916e6d32ff8.png)

.... although Windows is not happy about it.  

![image](https://user-images.githubusercontent.com/5569508/83358685-1422cb80-a32a-11ea-9015-beede90c2b92.png)


Linux, of course, works just fine with trailing whitespace.

**Comment by IanMercer** [comment]
@Clockwork-Muse Good point. My issue is the inconsistency in the `Enumerate` call: the search part succeeds because it trims the path, but the resulting filename fails because it doesn't. 

Given that Linux allows directory names with a space on the end, the `Enumerate` call should probably fail to find `C:\<space>` rather than trimming it and returning bogus results. As coded, you could never Enumerate the files in a Linux directory with a space on the end of the directory name. Maybe that's the real bug here?

**Comment by Clockwork-Muse** [comment]
Probably.

**Comment by Jozkee** [comment]
We probably want to investigate why the `path` in `EnumerateFiles` is being trimmed. If we decide to make the change we should consider that not trimming the whitespace on a next version could be considered breaking.

**Comment by elgonzo** [comment]
> not trimming the whitespace on a next version could be considered breaking

That's true. However, currently there doesn't seem a way to access files or directories with a trailing whitespace through System.IO functionality (if there is any clever trick to do so without resorting to p/invoke native host APIs, i haven't found it yet). 

If the current System.IO APIs remain unchanged in their behavior (which would not be a bad move), then there should be _some_ way to enumerate and access such files and directories in a straightforward manner, in my opinion. If such functionality is not a candidate to be incorporated into the BCL, perhaps a platform extension might be a fitting avenue to offer robust crossplatform functions for file/directory access.
